crates/oxc_linter/src/rules/eslint/no_import_assign.rs

```diff
@@ -154,19 +189,18 @@ fn test() {
         ("import * as mod9 from 'mod'; ({ bar: mod9 } = foo)", None),
         ("import * as mod10 from 'mod'; ({ bar: mod10 = 0 } = foo)", None),
         ("import * as mod11 from 'mod'; ({ ...mod11 } = foo)", None),
-        // TODO
-        // ("import * as mod1 from 'mod'; mod1.named = 0", None),
-        // ("import * as mod2 from 'mod'; mod2.named += 0", None),
-        // ("import * as mod3 from 'mod'; mod3.named++", None),
-        // ("import * as mod4 from 'mod'; for (mod4.named in foo);", None),
-        // ("import * as mod5 from 'mod'; for (mod5.named of foo);", None),
-        // ("import * as mod6 from 'mod'; [mod6.named] = foo", None),
-        // ("import * as mod7 from 'mod'; [mod7.named = 0] = foo", None),
-        // ("import * as mod8 from 'mod'; [...mod8.named] = foo", None),
-        // ("import * as mod9 from 'mod'; ({ bar: mod9.named } = foo)", None),
-        // ("import * as mod10 from 'mod'; ({ bar: mod10.named = 0 } = foo)", None),
-        // ("import * as mod11 from 'mod'; ({ ...mod11.named } = foo)", None),
-        // ("import * as mod12 from 'mod'; delete mod12.named", None),
+        ("import * as mod1 from 'mod'; mod1.named = 0", None),
+        ("import * as mod2 from 'mod'; mod2.named += 0", None),
+        ("import * as mod3 from 'mod'; mod3.named++", None),
+        ("import * as mod4 from 'mod'; for (mod4.named in foo);", None),
+        ("import * as mod5 from 'mod'; for (mod5.named of foo);", None),
+        ("import * as mod6 from 'mod'; [mod6.named] = foo", None),
+        ("import * as mod7 from 'mod'; [mod7.named = 0] = foo", None),
+        ("import * as mod8 from 'mod'; [...mod8.named] = foo", None),
+        ("import * as mod9 from 'mod'; ({ bar: mod9.named } = foo)", None),
+        ("import * as mod10 from 'mod'; ({ bar: mod10.named = 0 } = foo)", None),
+        ("import * as mod11 from 'mod'; ({ ...mod11.named } = foo)", None),
+        ("import * as mod12 from 'mod'; delete mod12.named", None),
```

Comment:
The import of these cases all belong to the `ImportNamespaceSpecifier`, so could we try to limit the check?

Reply:
AstKind don't implement `ImportNamespaceSpecifier` could you help me how can i use it?

Reply:
Yes. We need to finish #2013

Reply:
ok, i'll follow this PR, i think with these changes i can solve the property check problem, alright?

Reply:
Yes

Reply:
![image](https://github.com/oxc-project/oxc/assets/46359732/fe77abb9-1905-4495-a5e3-cecca3c66f6c)

i can't solve this because these case is `ImportNamespaceSpecifier` also and any approach over in this case will decrease significatively the linter performance.